humility-net-core/src/lib.rs

```diff
@@ -239,7 +282,18 @@ impl NetCore {
         udp_dump.reinitialize_dump_from(dump_index)?;
 
         // By construction, this DumpAgentCore has exactly what it needs!
-        agent_core.read_8(addr, data)?;
+        let data_len = data.len();
+        agent_core
+            .read_8(addr, &mut data[..(region.size as usize).min(data_len)])?;
+
+        // If the buffer is larger than this region, then recurse in an attempt
+        // to keep reading data.
+        if overflow {
+            self.read_ram(
```

Comment:
I have no context for how many times this might recurse - do we need to worry about blowing up the stack?

Reply:
Probably not – this only matters for the task table, which is ~30 items.

However, I agree that it's janky, so I changed it to a loop anyhow: c93a47529eb30c726814a4d7e9d5cb58b1f881cf